staging/src/k8s.io/kubectl/pkg/cmd/config/unset.go

```diff
@@ -33,25 +35,31 @@ import (
 type unsetOptions struct {
 	configAccess clientcmd.ConfigAccess
 	propertyName string
+	jsonPath     bool
+
+	streams genericclioptions.IOStreams
 }
 
 var (
 	unsetLong = templates.LongDesc(i18n.T(`
 	Unset an individual value in a kubeconfig file.
 
-	PROPERTY_NAME is a dot delimited name where each token represents either an attribute name or a map key.  Map keys may not contain dots.`))
+	PROPERTY_NAME is either a jsonpath query for where the property name will be, or a dot delimited name.`))
 
 	unsetExample = templates.Examples(`
 		# Unset the current-context
 		kubectl config unset current-context
 
 		# Unset namespace in foo context
-		kubectl config unset contexts.foo.namespace`)
+		kubectl config unset contexts.foo.namespace
+
+		# Unset cluster server using jsonpath
+		kubectl config unset '{.clusters[?(@.name=="cluster-0")].cluster.server}' --jsonpath`)
```

Comment:
Is this the correct usage? In `set.go` it appears the `--jsonpath` flag is not needed?

Reply:
Whoops thanks for the catch.